### PR TITLE
feat(scanv4): Fallback if no default StorageClass

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

Implement fallback for environments not having a default StorageClass configured.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: image/templates/helm/shared/templates/_storage_classes.tpl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  {{ $_ := set $._rox.env.storageClasses "all" (dict "all" $storageClasses) }}
+
+  {{- range $storageClassName, $storageClassProperties := $storageClasses -}}
+    {{/* Would like to use `break`, but there are quite a few Helm 3.x versions out there not supporting that...
+        On the other hand there should, of course, only be at most one default storage class on a cluster. ¯\_(ツ)_/¯ */}}
+    {{- if and (get $storageClassProperties "isDefault") (not $defaultStorageClass) -}}
+      {{- $defaultStorageClass = $storageClassName -}}
+    {{- end -}}
+  {{- end -}}
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply image/templates/helm/shared/templates/_storage_classes.tpl:30]: You are right, this is unnecessary, I will simplify this. Thanks!

File: pkg/helm/charts/tests/centralservices/testdata/helmtest/suite.yaml [hunk 2]
```diff
@@ -29,6 +29,17 @@ defs: |
...
+  objects:
+    # Apparently, for the `lookup` function in the Helm chart to be able to list the objects of a certain kind
+    # at least on such object needs to exist. Therefore, we create a StorageClass here solely
+    # for enabling lookups in the charts under test.
+    - apiVersion: storage.k8s.io/v1
+      kind: StorageClass
+      metadata:
+        name: some-default-sc
+        annotations:
+          storageclass.kubernetes.io/is-default-class: false
```
[the remark this answers is not in the document]
Comment (mclasmeier) [reply pkg/helm/charts/tests/centralservices/testdata/helmtest/suite.yaml]: The joy of copy and paste. :-D
Thanks.

File: image/templates/helm/shared/templates/_storage_classes.tpl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {{- $storageClassName := $sc.metadata.name -}}
+    {{- $annotations := $sc.metadata.annotations | default dict -}}
+    {{- $isDefault := index $annotations "storageclass.kubernetes.io/is-default-class" | default false -}}
+    {{- $_ := set $storageClasses $storageClassName (dict "isDefault" $isDefault) -}}
+    {{- if and $isDefault (not $defaultStorageClass) -}}
+      {{- $defaultStorageClass = $storageClassName -}}
+    {{- end -}}
+  {{- end -}}
+
+  {{ $_ := set $._rox.env.storageClasses "all" (dict "all" $storageClasses) }}
```
Comment (johannes94) [inline image/templates/helm/shared/templates/_storage_classes.tpl]: Maybe I'm missing something but why do we need to add the "all" field here and in the config shape. I don't see where it is being used in this PR.

Comment (mclasmeier) [reply image/templates/helm/shared/templates/_storage_classes.tpl]: Felt more complete to me.
But you are right, it is currently not used.
I'm open for removal.

Comment (mclasmeier) [reply image/templates/helm/shared/templates/_storage_classes.tpl]: discussed offline. Might come in handy in the future to have knowledge about existing storageclasses.

**Review by johannes94 — APPROVED** [review]
Feedback from @dcaravel was addressed, I reviewed that change. Looks good to me.

I'm fine also fine with leaving the `all` field as is, could be useful if we add more validation logic later on.

**Pull request merged**